tests/test_unmanaged_resources.py

```diff
@@ -0,0 +1,129 @@
+        {"unmanaged_resource_id": "test::Resource[agent1,key4=key4],v=1", "values": {"value1": "test7", "value2": "test8"}},
+        {"unmanaged_resource_id": "test::Resource[agent1,key5=key5],v=1", "values": {"value1": "test9", "value2": "test10"}},
+        {"unmanaged_resource_id": "test::Resource[agent1,key6=key6],v=1", "values": {"value1": "test11", "value2": "test12"}},
+    ]
+    result = await client.unmanaged_resource_create_batch(environment, resources)
+    assert result.code == 200
+
+    result = await client.unmanaged_resources_get_batch(
+        environment, start_resource_id="aaa", end_resource_id="test::Resource[agent1,key3=key3],v=1", limit=2
+    )
+    assert result.code == 400
+    assert "the start_resource_id is not formatted" in result.result["message"]
+
+    result = await client.unmanaged_resources_get_batch(
+        environment, start_resource_id="test::Resource[agent1,key1=key1],v=1", end_resource_id="aaa", limit=2
+    )
+    assert result.code == 400
+    assert "the end_resource_id is not formatted" in result.result["message"]
+
+    # will not get more than limit resources
+    result = await client.unmanaged_resources_get_batch(environment, limit=2)
+    assert result.code == 200
+    assert len(result.result["data"]) == 2
+    for i in range(0, 2):
+        assert result.result["data"][i]["unmanaged_resource_id"] == resources[i]["unmanaged_resource_id"]
+        assert result.result["data"][i]["values"] == resources[i]["values"]
+
+    # will not get resources before start_resource_id
+    result = await client.unmanaged_resources_get_batch(
+        environment, start_resource_id="test::Resource[agent1,key3=key3],v=1", limit=3
+    )
+    assert result.code == 200
+    assert len(result.result["data"]) == 3
+    for i in range(3, 6):
+        assert result.result["data"][i - 3]["unmanaged_resource_id"] == resources[i]["unmanaged_resource_id"]
+        assert result.result["data"][i - 3]["values"] == resources[i]["values"]
+
+    # will not get resources after end_resource_id
+    result = await client.unmanaged_resources_get_batch(
+        environment,
+        start_resource_id="test::Resource[agent1,key2=key2],v=1",
+        end_resource_id="test::Resource[agent1,key4=key4],v=1",
+        limit=3,
+    )
+    assert result.code == 200
+    assert len(result.result["data"]) == 1
+    assert result.result["data"][0]["unmanaged_resource_id"] == resources[2]["unmanaged_resource_id"]
+    assert result.result["data"][0]["values"] == resources[2]["values"]
+
+
+async def test_discovery_resource_bad_res_id(server, client, agent, environment):
+    """
+    Test that exceptions are raised when creating unmanaged resources with invalid IDs.
+    """
+    result = await client.unmanaged_resource_create(environment, "test", {"value1": "test1", "value2": "test2"})
+    assert result.code == 500
```

Comment:
wrong input from a user should results in a 400. Or at least something in the range 4xx. 500 means that the server is causing the problem, which is not the case here.